client/me/notification-settings/reader-subscriptions/index.jsx

```diff
@@ -41,9 +47,23 @@ class NotificationSubscriptions extends Component {
 	}
 
 	handleCheckboxEvent( action, invert = false ) {
-		return ( event ) => {
-			const optionValue = invert ? ! event.target.checked : event.target.checked;
-			this.props.recordGoogleEvent( 'Me', `Clicked ${ action } checkbox`, 'checked', +optionValue );
+		return ( checked ) => {
+			const optionValue = invert ? ! checked : checked;
+			// Create a synthetic event object that matches what updateSetting expects
+			const syntheticEvent = {
+				currentTarget: {
+					name: action,
+					value: optionValue,
+				},
+			};
+			this.props.toggleSetting( syntheticEvent );
+			const actionLabel = this.checkboxNameToActionMap[ action ];
+			this.props.recordGoogleEvent(
+				'Me',
+				`Clicked ${ actionLabel } checkbox`,
+				'checked',
+				+optionValue
+			);
```

Comment:
Can you explain a bit more what's going on here with syntheticEvent (ie, why that's needed)?

Reply:
Indeed!

Follow up on props.toggleSetting and you'll find the handler expects an event, but CheckboxControl [simplifies the checkbox onChange](https://github.com/WordPress/gutenberg/blob/1eb798d790ee7cb92199ac80f1bb463cdd464ac9/packages/components/src/checkbox-control/index.tsx#L92) handler to a simple `boolean` parameter (checked vs unchecked, true/false).

So, to keep the functionality without going on a refactor adventure upstream, I saw the same strategy was already in use on `handleModalCancel` (around line 68) and decided to go for it.

Reply:
Thanks for the explanation. And indeed, we do already use this pattern. I also just dug around a bit to understand how this.props.toggleSetting() works, and what we're doing here seems to make sense.

Reply:
If anyone else wonders... I was wondering why we didn't need such special treatment in other places where we update to use CheckboxControl, but in those cases, we are already separately defined event handlers for the components. In this component, the event handler is passed in via the 'withFormBase' HOC.